test/Microsoft.AspNet.Mvc.DataAnnotations.Test/DataAnnotationsModelValidatorProviderTest.cs

```diff
@@ -39,6 +39,29 @@ public void GetValidators_ReturnsValidatorForIValidatableObject()
 #endif
 
         [Fact]
+        public void GetValidators_InsertsRequiredValidatorsFirst()
+        {
+            var provider = new DataAnnotationsModelValidatorProvider(
+                new TestOptionsManager<MvcDataAnnotationsLocalizationOptions>(),
+                stringLocalizerFactory: null);
+            var metadata = _metadataProvider.GetMetadataForProperty(
+                typeof(ClassWithProperty),
+                "PropertyWithMultipleValidationAttributes");
+
+            var providerContext = new ModelValidatorProviderContext(metadata);
+
+            // Act
+            provider.GetValidators(providerContext);
+
+            // Assert
+            Assert.Equal(4, providerContext.Validators.Count);
```

Comment:
```C#
Assert.Collection(providerContext.Validators,
  validator => 
  {
      var dataAnnotationsValidator = Assert.IsType<DataAnnotationsModelValidator>(validator);
      Assert.IsType<CustomRequiredAttribute2>(dataAnnotationsValidator);
  },
 {
      var dataAnnotationsValidator = Assert.IsType<DataAnnotationsModelValidator>(validator);
      Assert.IsType<CustomRequiredAttribute1>(dataAnnotationsValidator);
  },
  ...
  );
```

Verify this works on Mono since this depends on reflection order so internal ordering might be platform dependent.